### PR TITLE
Stop execution after delegatecall suicide

### LINKED ISSUE #876: Delegatecall Double Suicide
Looking at the graph, of contracts with a delegatecall to a suicide instruction, mythril doesn't seem to realize that transaction execution is supposed to stop after a selfdestruct.
```
pragma solidity ^0.4.25;

contract Vulnerable {
  uint256 a;
  Callee private callee;

  constructor(address _callee) public payable {
    callee = Callee(_callee);
  }

  function kill(uint256 n) public {
    callee.delegatecall(bytes4(sha3("abc(uint256)")), n);
    // Unreachable code:
    selfdestruct(msg.sender); // Did you know that you could kill yourself twice?
  }
}

contract Callee {
    uint256 a;
  function abc(uint256 n) public returns(uint256) {
    selfdestruct(msg.sender); // Did you know that you could kill yourself twice?
  }
}
```

### PR DESCRIPTION
**Pull request by nbanmp**
Fixes #876 

This took a lot more work than it should have taken... but it should be good now.

### CODE REVIEW

**Comment by JoranHonig** [comment]
If a suicide happens in a delegate call, then the caller is scheduled for suicide right? Shouldn't execution still continue?

**Comment by norhh** [comment]
the corresponding `account.deleted` becomes `true` in that case, so what @nbanmp is doing might be useful(or maybe not as i am not sure of the behavior in that case) when there are multiple calls and the code is in some call depth >2. when depth is 1 in that case the account gets deleted(pruned by that loop) and execution stops.

**Comment by nbanmp** [comment]
@JoranHonig I'm not entirely sure of the correct behavior in that case. My tests with the following contract in ganache did not show continued execution.
```
pragma solidity ^0.4.25;

contract Vulnerable {
    uint256 a;
    Callee private callee;
    Callee2 private callee2;

    constructor(address _callee, address _callee2) public payable {
        callee = Callee(_callee);
        callee2 = Callee2(_callee2);
    }

    function go() public {
        callee.delegatecall(bytes4(sha3("abc2()")));
    }
}

contract Callee {
    uint256 a;
    Callee2 private callee;
    Callee2 private callee2;

    constructor(address _callee) public payable {
        //callee = Callee2(_callee);
    }

    function abc2() public returns(uint256) {
        selfdestruct(msg.sender);
        callee2.call(bytes4(sha3("abc()")));
    }
}

contract Callee2 {
    uint256 public a;
    function abc() public returns(uint256) {
        a = 1337;
    }

}
```

@norhh Yeah, I spent way too long making different call depths work.

File: mythril/laser/ethereum/instructions.py [hunk 1]
```diff
@@ -1627,13 +1627,6 @@ def suicide_(self, global_state: GlobalState):
                 )
                 account_created = True
 
-        global_state.environment.active_account = deepcopy(
-            global_state.environment.active_account
```
Comment (JoranHonig) [inline mythril/laser/ethereum/instructions.py]: Isn't this required because the environment is shallow copied?

**Comment by JoranHonig** [comment]
@nbanmp looking at this again after a while.
````
pragma solidity ^0.4.25;

contract Vulnerable {
    uint256 a;
    Callee private callee;
    Callee2 private callee2;

    constructor(address _callee, address _callee2) public payable {
        callee = Callee(_callee);
        callee2 = Callee2(_callee2);
    }

    function go() public {
        callee.delegatecall(bytes4(sha3("abc2()")));
        suicide(msg.sender);
    }
}
```

Should self-destruct the contract twice (the second doesn't really affect anything), at least that is what seems should happen if I look at the yellowpaper & geth.

**Comment by nbanmp** [comment]
In that case, I'll close this.

**Pull request closed without merging**